src/ngx_http_lua_socket_tcp.c

```diff
@@ -2462,6 +2462,20 @@ ngx_http_lua_socket_tcp_send(lua_State *L)
             len = ngx_http_lua_calc_strlen_in_table(L, 2, 2, 1 /* strict */);
             break;
 
+        case LUA_TNIL:
+            len = sizeof("nil") - 1;
+            break;
+
+        case LUA_TBOOLEAN:
+            if (lua_toboolean(L, 2)) {
+                len = sizeof("true") - 1;
+
+            } else {
+                len = sizeof("false") - 1;
+            }
+
+            break;
```

Comment:
Sending booleans and nils are more likely due to bugs in the application code? I'm not sure if this is the right thing to do.

Reply:
@agentzh
It's for the sake of duck type. API like `ngx.log` and `ngx.say` handle boolean and nil by converting them to string. If we finally migrate cosocket to FFI implementation, we will use `tostring` to non-string type argument too. BTW, it also conforms the description in the error message: "string, number, boolean, nil, or array table expected".

Reply:
@spacewander OK then.